test/Library-test.js

```diff
@@ -106,24 +112,103 @@ describe("library.js", function() {
 
-      assert.equal(error2, "Sorry, there are currently no copies of Yes Please available at the Denver Public Library");
+      assert.equal(error2, "Sorry, there are currently no copies of Yes Please available at the Denver Public Library.");
+    });
+  });
+
+  describe('takeStock', function() {
+
+    it.skip('should be able to take stock of how many books are on a shelf', function() {
+      var hyperion = {
+        title: "Hyperion",
+        mainCharacter: { name: "The Shrike", age: null, pronouns: "they" },
+        pageCount: 482,
+        genre: "fantasy"
+      };
+
+      var dune = {
+        title: "Dune",
+        mainCharacter: { name: "Paul", age: 15, pronouns: "he/him" },
+        pageCount: 421,
+        genre: "fantasy"
+      };
+
+      var prideAndPrejudice = {
+        title: "Pride and Prejudice",
+        mainCharacter: { name: "Eliabeth Bennet", age: 20, pronouns: "she/her" },
+        pageCount: 432,
+        genre: "fiction"
+      }
+
+      var auroraLibrary = createLibrary("Aurora Public Library");
+
+      addBook(auroraLibrary, hyperion);
+      addBook(auroraLibrary, dune);
+      addBook(auroraLibrary, prideAndPrejudice);
+
+      assert.equal(takeStock(auroraLibrary, 'fantasy'), "There are a total of 2 fantasy books at the Aurora Public Library.")
+    });
+
+    it.skip('should be able to count the number of books on a different shelf', function() {
+      var bornACrime = {
+        title: "Born a Crime",
+        mainCharacter: { name: "Trevor Noah", age: 36, pronouns: "he/him" },
+        pageCount: 304,
+        genre: "nonFiction"
+      };
+
+      var laLibrary = createLibrary("Los Angeles Public Library");
+
+      addBook(laLibrary, bornACrime);
+      assert.equal(takeStock(laLibrary, 'nonFiction'), "There are a total of 1 nonFiction books at the Los Angeles Public Library.")
+    });
+
+    it.skip('should be able to take stock of how many books are in the entire collection if a specific shelf is not specified', function() {
+      var dracula = {
+        title: "Dracula",
+        mainCharacter: { name: "Count Dracula", age: undefined, pronouns: "he/him" },
+        pageCount: 418,
+        genre: "fantasy"
+      };
+
+      var bornACrime = {
+        title: "Born a Crime",
+        mainCharacter: { name: "Trevor Noah", age: 36, pronouns: "he/him" },
+        pageCount: 304,
+        genre: "nonFiction"
+      };
+
+      var prideAndPrejudice = {
+        title: "Pride and Prejudice",
+        mainCharacter: { name: "Eliabeth Bennet", age: 20, pronouns: "she/her" },
+        pageCount: 432,
+        genre: "fiction"
+      };
+
+      var denverLibrary = createLibrary("Denver Public Library");
+
+      addBook(denverLibrary, dracula);
+      addBook(denverLibrary, bornACrime);
+      addBook(denverLibrary, prideAndPrejudice);
+
+      assert.equal(takeStock(denverLibrary), "There are a total of 3 books at the Denver Public Library.")
```

Comment:
I like this extra bit of spice to make sure that they have conditional logic if a specific argument hasn't been passed through.  Sounds like it will be a great opportunities for students to practice refactoring.  Looks great Kayla!